src/traffic_cop/traffic_cop.cpp

```diff
@@ -364,6 +366,16 @@ TrafficCopResult TrafficCop::CodegenPhysicalPlan(
       common::ManagedPointer<const std::string>(&portal->GetStatement()->GetQueryText()));
 
   // TODO(Matt): handle code generation failing
+  auto &resource_metrics = common::thread_context.resource_tracker_.GetMetrics();
+
+  const bool query_trace_metrics_enabled =
+      common::thread_context.metrics_store_ != nullptr &&
+      common::thread_context.metrics_store_->ComponentToRecord(metrics::MetricsComponent::QUERY_TRACE);
+  if (query_trace_metrics_enabled) {
+    common::thread_context.metrics_store_->RecordQueryText(exec_query->GetQueryId(),
+                                                           portal->GetStatement()->GetQueryText(), resource_metrics);
```

Comment:
We don't need the `resource_metrics` for query texts. We can just remove that one. Instead, we can append the timestamp at the end.

Reply:
Removed `resource_metrics`.